startUp.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+
+import sys
+import subprocess
+import time
+#process names from text file stored into array
+#sleep after each check for 2 secs
+
+
+def main(): 
+    #read program names from file and put them into an array
+    fileName = open(sys.argv[1])
+    programs = fileName.read().splitlines()
+    fileName.close()
+    
+    #start up programs in separate processes
+    p0 = subprocess.Popen(programs[0])
+    p1 = subprocess.Popen(programs[1])
+    '''
+    p2 = subprocess.Popen(programs[2])
+    p3 = subprocess.Popen(programs[3])
+    '''
+    
+    MAXRESTART = 10
+    timesRestarted = [0, 0, 0, 0]
+    
+    #check processes and respond accordingly
+    while True:
+        if(p0.poll() is not None):
+            #print(programs[0], "died")
+            p0 = subprocess.Popen(programs[0])
+            timesRestarted[0] += 1
+            if (timesRestarted[0] > MAXRESTART):
+                print(programs[0], "exceeded", MAXRESTART, "restarts")
+                return
+        if(p1.poll() is not None):
+            #print(programs[1], "died")
+            p1 = subprocess.Popen(programs[1])
+            timesRestarted[1] += 1
+            if (timesRestarted[1] > MAXRESTART):
+                print(programs[1], " exceeded", MAXRESTART, " restarts")
+                return     
```

Comment:
You can see the code for p1 and p2 are almost identical, I'm sure it's possible to shorten the code by making it a loop for any number of processes :)